mk/spksrc.plist.mk

```diff
@@ -9,14 +9,13 @@ endif
 
 .PHONY: cat_PLIST
 cat_PLIST:
-	@for depend in $(DEPENDS) ; \
+	@echo Processing PLIST dependencies 1>&2
+	@for depend in `$(MAKE) dependency-list` ; \
```

Comment:
If you take dependency-list instead of `$(DEPENDS)` you get the dependencies of  `$(BUILD_DEPENDS) $(DEPENDS) $(OPTIONAL_DEPENDS)`
but this will fail because all files in PLIST must exist (and the optional are dependencies that exist not for all archs).
And BUILD_DEPENDS is for packages that are used only in the build environment and not needed at runtime (i.e. dependency of another spk like python).
If dependency-list fixes the PLIST file for some packages, those packages may have used BUILD_DEPENDS but should have used DEPENDS.
As for fossil-scm if the error you meant in #4222 is missing PLIST files of openssl or zlib, than the real error is that fossil-scm has defined BUILD_DEPENDS instead of DEPENDS for cross/zlib and cross/openssl.